src/clientenvironment.cpp

```diff
@@ -207,6 +208,9 @@ void ClientEnvironment::step(float dtime)
 
 	//std::cout<<"Looped "<<loopcount<<" times."<<std::endl;
 
+	GenericCAO *playercao = lplayer->getCAO();
+	bool player_immortal = playercao ? playercao->isImmortal() : false;
```

Comment:
bool player_immortal = playercao && playercao->isImmortal();

Reply:
That was what I wrote originally, others suggested it looks ugly to call getCAO() twice.  
Will change to that if you are sure.

Reply:
I mean, I called getCAO directly before, I'll switch

Reply:
if it's inlined it's not problematic for me and remove a useless temporary variable :p

Reply:
Done :)